### ISSUE TITLE
Cursor position adds empty rows or columns to csv files

### ISSUE THREAD

**Issue by yamanq** [andmarti1424/sc-im#457]
## Compiler used and options (although Makefile would be welcome)
Used the Makefile that was provided with the project and the standard options for building on Linux.

## Actions done
1. Open sc-im with either a csv file or just a blank spreadsheet
2. Add values to the spreadsheet
3. Move the cursor past last data value in either direction (rows or columns)
4. Save back to csv file

## sc-im behavior
Adds unexpected empty lines and/or commas. For example, if you input a number on A1 and move the cursor down to C5, there would be empty rows/columns after you save. If you move the cursor back up before saving, those empty rows/columns would disappear.

## Expected behavior
Moving the cursor should not add empty rows/columns in csv files, especially if this only depends on the location of the cursor at the time of saving.

**Comment by andmarti1424** [comment]
Yes, you can call this a bug. Btw, you can always select the range you want to export and do  it with
:e csv

**Comment by andmarti1424** [comment]
@yamanq Could you please update to latest commit and try again? Thanks.

**Comment by yamanq** [comment]
@andmarti1424 Thanks for working on this. It seems that the issue is fixed with extra rows, but extra columns are still added if the cursor is moved based the last column.

**Comment by andmarti1424** [comment]
@yamanq Could you please update to latest commit and try again? Thanks!!

**Comment by yamanq** [comment]
@andmarti1424 I tried from a787115fbeb3379d707a594227acd2330d4b21ad and it results in some bugs in how the file is saved. I opened a csv file with 2 rows and 3 columns. Any element added past the third column is not saved and any element added past the second row results in the third **column** being deleted. Also, just placing the cursor on the third row and then saving results in the third column being deleted.
